digit-recognition/frontend/src/lib/hooks/use-media.ts

```diff
@@ -0,0 +1,58 @@
+import { useEffect, useState } from "react";
+
+/**
+ * Hook that tracks the state of a CSS media query.
+ */
+export function useMedia(query: string) {
```

Comment:
I'm against matching queries in JS, quite a lot of complexity, performance impact and sometimes unpredictable behavior. CSS media queries are much more reliable.